### PR TITLE
[chore] Fixed response for errors in project storage collection filters

### PR DESCRIPTION
**Pull request by Kharonus**
- added request spec for project storages endpoint
- added human name to filters
- added permission check for GET project storage by id

### CODE REVIEW

**Review by dombesz — APPROVED** [review]
LGTM!  I just left one question, but feel free to merge.

File: modules/storages/app/models/queries/storages/project_storages/filter/storage_id_filter.rb [hunk 1]
```diff
@@ -34,6 +34,10 @@ def type
       :list
     end
 
+    def human_name
+      ::Storages::ProjectStorage.human_attribute_name(name)
```
Comment (dombesz) [inline modules/storages/app/models/queries/storages/project_storages/filter/storage_id_filter.rb:38]: Should we also add this method to the [`modules/storages/app/models/queries/storages/work_packages/filter/storages_filter_mixin.rb`](https://github.com/opf/openproject/blob/fb0841dd7deb2054c8f388dbb1c6a73a595367df/modules/storages/app/models/queries/storages/work_packages/filter/storages_filter_mixin.rb) ?

**Comment by Kharonus** [comment]
⚠️ There is a test case missing, which is still red. found it by chance: 

If there is a filter set for project id, and the project exists, but does not contain any project_storages, for some reason, the query also goes into error mode. Don't know why yet, added a test case and need to make it greenn.

File: modules/storages/app/models/queries/storages/project_storages/filter/storage_id_filter.rb (hunk 1 above)
Comment (Kharonus) [reply modules/storages/app/models/queries/storages/project_storages/filter/storage_id_filter.rb:38]: true, those probably also fail with an ugly 501 error, when receiving a wrong filter value. The `human_name` method is called on a filter, when the query tries to build error messages for them.

Comment (Kharonus) [reply modules/storages/app/models/queries/storages/project_storages/filter/storage_id_filter.rb:38]: added

**Comment by Kharonus** [comment]
Fixed the issue, it was the wrong result in `allowed_values`. Can you check again, @dombesz ?

**Review by dombesz — APPROVED** [review]

**Pull request merged**